### PR TITLE
Adding cover art retrieval for Subsonic.

### PR DESCRIPTION
**Pull request by davidgfnet**
You guys probably want some sort of mechanism to generalize this hack :D

### CODE REVIEW

File: src/covers/albumcoverloader.h [hunk 1]
```diff
@@ -104,11 +105,12 @@ class AlbumCoverLoader : public QObject {
   QQueue<Task> tasks_;
   QMap<QNetworkReply*, Task> remote_tasks_;
   QMap<QString, Task> remote_spotify_tasks_;
+  QMap<QString, Task> remote_subsonic_tasks_;
   quint64 next_id_;
 
   NetworkAccessManager* network_;
 
-  bool connected_spotify_;
+  bool connected_spotify_, connected_subsonic_;
```
Comment (hatstand) [inline src/covers/albumcoverloader.h]: One variable per line

File: src/internet/subsonic/subsonicservice.cpp [hunk 2]
```diff
@@ -494,6 +495,15 @@ void SubsonicLibraryScanner::OnGetAlbumFinished(QNetworkReply* reply) {
   }
 }
 
+void SubsonicLibraryScanner::OnGetAlbumCoverFinished(QNetworkReply* reply, QString albumid) {
+  reply->deleteLater();
+  pending_requests_.remove(reply);
+
+  QByteArray image = reply->readAll();
+
+  service_->emitImageLoaded(albumid, QImage::fromData(image));
```
Comment (hatstand) [inline src/internet/subsonic/subsonicservice.cpp]: Don't do this. Instead add a signal to `SubsonicLibraryScanner` and connect that signal directly to the `ImageLoaded` signal in `SubsonicService`.

Comment (davidgfnet) [inline src/internet/subsonic/subsonicservice.cpp]: Yeah I knew you'd say that, but again, hack over hack.
Do you guys plan to rework the cover art links? We could create URI handlers so services can hook them, more generic and abstract. What do you think?

Comment (hatstand) [inline src/internet/subsonic/subsonicservice.cpp]: Yeah, that would be much nicer.

**Pull request closed without merging**